### PR TITLE
Fix decimal handling of ETH/ERC20 for deposit

### PR DESCRIPTION
**Pull request by veado**
- [x] Fix decimal handling of ETH/ERC20 for deposit
- [x] Add `convertBaseAmountDecimal` (incl. tests)
- [x] Remove `convertToBase8` in favour of `convertBaseAmountDecimal`
- [x] Use `convertBaseAmountDecimal` in `getSwapResult` + `getSlip`
- [x] Update misc. helpers for using object as params
- [x] Change `Record<string, PoolData>` to `PoolsDataMap`

### CODE REVIEW

File: src/renderer/helpers/assetHelper.ts [hunk 1]
```diff
@@ -163,3 +165,17 @@ export const midgardAssetFromString: (assetString: string) => O.Option<Asset> =
...
+/**
+ * Helper to convert asset amounts from Midgard/THORChain based on 1e8 decimal
+ * into any 1e(n) decimal based amount
+ *
+ * For example:
+ * ETH.ETH: Midgard 1e8 vs. ETH 1e18
+ * @param amount Asset amount to convert
+ * @param decimal Asset decimal
+ */
+export const convertDecimalFromMidgard = (amount: BigNumber, decimal: number): BigNumber => {
```
Comment (veado) [inline src/renderer/helpers/assetHelper.ts:178]: ^ change function to
```ts
(amount: BaseAmount) => BaseAmount
```

File: src/renderer/helpers/assetHelper.ts [hunk 2]
```diff
@@ -163,3 +165,17 @@ export const midgardAssetFromString: (assetString: string) => O.Option<Asset> =
   // And 0X isn't recognized as valid address in ethers lib
   O.map(updateEthChecksumAddress)
 )
+
+/**
+ * Helper to convert asset amounts from Midgard/THORChain based on 1e8 decimal
+ * into any 1e(n) decimal based amount
+ *
+ * For example:
+ * ETH.ETH: Midgard 1e8 vs. ETH 1e18
```
Comment (veado) [inline src/renderer/helpers/assetHelper.ts:174]:
```
Example ETH.ETH: 
- Midgard 1e8 decimal (for any asset)
- ETH 1e18 decimal

In this case `convertDecimalFromMidgard` converts amount from 1e8 to 1e18 decimal
```

File: src/renderer/components/swap/Swap.tsx [hunk 3]
```diff
@@ -622,19 +626,25 @@ export const Swap = ({
       targetAsset,
       O.map((asset) => {
         const chainAsset = getChainAsset(asset.chain)
-        const chainAssetPoolData: PoolData | undefined = poolData[assetToString(chainAsset)]
-        const assetPoolData: PoolData | undefined = poolData[assetToString(asset)]
+        const chainAssetPoolData: PoolData | undefined = poolsData[assetToString(chainAsset)]
+        const assetPoolData: PoolData | undefined = poolsData[assetToString(asset)]
```
Comment (sarawut11) [inline src/renderer/components/swap/Swap.tsx:630]: We can use `O.fromNullable` with FP

**Review by sarawut11 — CHANGES_REQUESTED** [review]
looks good to me, just small comment

**Review by junkai121 — APPROVED** [review]
LGTM!

**Review by sarawut11 — APPROVED** [review]
LGTM!

**Pull request merged**